example/advanced/server-flex/advanced_server_flex.cpp

```diff
@@ -614,13 +521,53 @@ class http_session
         }
 
         // Send the response
-        handle_request(*doc_root_, parser_->release(), queue_);
+        enqueue(handle_request(*doc_root_, parser_->release()));
 
         // If we aren't at the queue limit, try to pipeline another request
-        if(! queue_.is_full())
+        if (response_queue_.size() < queue_limit)
             do_read();
     }
 
+    void
+    enqueue(http::message_generator response)
+    {
+        // Allocate and store the work
+        response_queue_.push_back(std::move(response));
+
+        // If there was no previous work, start the write
+        // loop
+        if (response_queue_.size() == 1)
+            do_write_loop();
+    }
+
+    // Called to start/continue the write-loop. Should not be called when
+    // write_loop is already active.
+    //
+    // Returns `true` if the caller may initiate a new read
+    bool
+    do_write_loop()
+    {
+        bool const was_full =
+            response_queue_.size() == queue_limit;
+
+        if (!response_queue_.empty()) {
+            http::message_generator gen =
+                std::move(response_queue_.front());
+            response_queue_.erase(response_queue_.begin());
+
+            bool keep_alive = gen.keep_alive();
+
+            beast::async_write(derived().stream(),
+                               std::move(gen),
+                               beast::bind_front_handler(
+                                   &http_session::on_write,
+                                   derived().shared_from_this(),
+                                   not keep_alive));
```

Comment:
we don't use the words for logical operators (`or`, `and`, etc) because 99% of people are not used to seeing it and it looks like an identifier. and the indent is too far